app/controllers/service_controller.rb

```diff
@@ -16,6 +16,16 @@ def show_list
     process_show_list(:dbname => :service, :gtl_dbname => :service)
   end
 
+  def show
+    if params[:display] == 'generic_objects'
+      super
+    else
+      super
+      @options = {}
+      @view, @pages = get_view(Vm, :parent => @record, :parent_method => :all_vms, :all_pages => true)
+    end
```

Comment:
I think this would be a little cleaner if we moved the call to super out of each branch since it is called in all conditions
```suggestion
    super
    if params[:display] != 'generic_objects'
      @options = {}
      @view, @pages = get_view(Vm, :parent => @record, :parent_method => :all_vms, :all_pages => true)
    end
```